postman/test_signing_server.go

```diff
@@ -0,0 +1,145 @@
+		return nil, err
+	}
+
+	return &types.Signature{
+		SigningPayload: &types.SigningPayload{
+			AccountIdentifier: &types.AccountIdentifier{Address: address},
+			Bytes:             bytes,
+			SignatureType:     types.EcdsaRecovery,
+		},
+		PublicKey: &types.PublicKey{
+			Bytes:     s.publicKey.Bytes(),
+			CurveType: types.Secp256k1,
+		},
+		SignatureType: types.EcdsaRecovery,
+		Bytes:         signatureBytes,
+	}, nil
+}
+
+func (s *signingServer) signHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "POST":
+		d := json.NewDecoder(r.Body)
+		signPayload := signRequest{}
+		err := d.Decode(&signPayload)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+
+		log.Printf("Received signing request with %d payloads\n", len(signPayload.Payloads))
+
+		var signatures []*types.Signature
+
+		for _, payload := range signPayload.Payloads {
+			signature, err := s.SignBytes(payload.AccountIdentifier.Address, payload.Bytes)
+			if err != nil {
+				http.Error(w, "unable to sign payload", 500)
+				return
+			}
+
+			signatures = append(signatures, signature)
+		}
+
+		resp := signResponse{Signatures: signatures}
+		bytes, err := json.Marshal(resp)
+		if err != nil {
+			http.Error(w, "unable to marshal response", 500)
+			return
+		}
+
+		_, _ = w.Write(bytes)
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
+}
+
+func (s *signingServer) run() {
+	http.HandleFunc("/sign", s.signHandler)
+
+	addr := fmt.Sprintf("localhost:%d", s.port)
+	log.Printf("Listening on %s\n", addr)
+
+	err := http.ListenAndServe(addr, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func parsePrivateKeyString(keyStr string) (*crypto.PrivateKeySECP256K1R, *crypto.PublicKeySECP256K1R, error) {
+	parts := strings.Split(keyStr, "-")
+	pkBytes, err := cb58.Decode(parts[1])
+	if err != nil {
+		return nil, nil, err
+	}
+	factory := crypto.FactorySECP256K1R{}
+	keyIntf, err := factory.ToPrivateKey(pkBytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pk := keyIntf.PublicKey()
+	return keyIntf.(*crypto.PrivateKeySECP256K1R), pk.(*crypto.PublicKeySECP256K1R), nil
+}
+
+func newSigningServer(port uint, privateKeyStr string) *signingServer {
+	privateKey, publicKey, err := parsePrivateKeyString(privateKeyStr)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return &signingServer{
+		privateKey: privateKey,
+		publicKey:  publicKey,
+		port:       port,
+	}
+}
+
+var opts struct {
+	port       uint
+	privateKey string
+}
+
+func init() {
+	flag.UintVar(&opts.port, "port", 9898, "signing server port")
+	flag.StringVar(&opts.privateKey, "private-key", "", "private key to sign with")
+	flag.Parse()
+}
+
+func main() {
+	println("!!! DO NOT USE THIS UTILITY WITH PRIVATE KEYS THAT HAS ACCESS TO REAL FUNDS!!!\n")
+	println("This is a test utility used to help testing Rosetta AVAX implementation during development\n\n")
+	signingServer := newSigningServer(opts.port, opts.privateKey)
+	signingServer.run()
+}
```

Comment:
Should this be removed?

Reply:
This is a test utility that I used together with the postman file attached. It allows one to execute the full construction flow through postman. You can see the README in the same folder on how it is used. 

I added it here based on @rajranjan0608's request as it helps with e2e testing during development. If the team things we shouldn't add, I can remove it.

Reply:
Signing helps in testing some of the endpoints through Postman. Should have no problem since we called it out in CAPS.